consent/strategy_default.go

```diff
@@ -1178,3 +1232,96 @@ func (s *DefaultStrategy) loginSessionFromCookie(r *http.Request) *flow.LoginSes
 
 	return ls
 }
+
+func (s *DefaultStrategy) requestDevice(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	return s.forwardDeviceRequest(ctx, w, r)
+}
+
+func (s *DefaultStrategy) forwardDeviceRequest(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	// Set up csrf/challenge/verifier values
+	verifier := strings.Replace(uuid.New(), "-", "", -1)
+	challenge := strings.Replace(uuid.New(), "-", "", -1)
+	csrf := strings.Replace(uuid.New(), "-", "", -1)
+
+	// Generate the request URL
+	iu := s.getDeviceVerificationPath(ctx)
+	iu.RawQuery = r.URL.RawQuery
+
+	deviceRequest := &flow.DeviceUserAuthRequest{
+		ID:          challenge,
+		Verifier:    verifier,
+		CSRF:        csrf,
+		RequestURL:  iu.String(),
+		RequestedAt: time.Now().Truncate(time.Second).UTC(),
+	}
+	f, err := s.r.ConsentManager().CreateDeviceUserAuthRequest(
+		r.Context(),
+		deviceRequest,
+	)
+	if err != nil {
+		return errorsx.WithStack(err)
+	}
+
+	encodedFlow, err := f.ToDeviceChallenge(ctx, s.r)
+	if err != nil {
+		return err
+	}
+	store, err := s.r.CookieStore(ctx)
+	if err != nil {
+		return err
+	}
+
+	CookieNameDeviceCSRF := s.r.Config().CookieNameDeviceCSRF(ctx)
+	if err := createCsrfSession(w, r, s.r.Config(), store, CookieNameDeviceCSRF, csrf, s.c.ConsentRequestMaxAge(ctx)); err != nil {
+		return errorsx.WithStack(err)
+	}
+
+	query := url.Values{"device_challenge": {encodedFlow}}
+	if r.URL.Query().Has("user_code") {
+		query.Add("user_code", r.URL.Query().Get("user_code"))
+	}
+
+	http.Redirect(w, r, urlx.SetQuery(s.c.DeviceVerificationURL(ctx), query).String(), http.StatusFound)
```

Comment:
as much as i'm contradicting myself with the comments above, `urlx.SetQuery(s.c.DeviceVerificationURL(ctx), query).String()` is really hard to read in there, try to indent like 

```
http.Redirect(
  w, 
  r, 
  urlx.SetQuery(s.c.DeviceVerificationURL(ctx), query).String(),
  http.StatusFound,
)
```

maybe it looks better, or maybe not